dnserver/main.py

```diff
@@ -173,3 +175,8 @@ def stop(self):
     @property
     def is_running(self):
         return (self.udp_server and self.udp_server.isAlive()) or (self.tcp_server and self.tcp_server.isAlive())
+
+    def add_record(self, **kwargs):
+        zone = Zone(**kwargs)
```

Comment:
Better to do:
```suggestion
    def add_record(self, zone: Zone):
```
So users get type hints.

Reply:
or you could add a `ZoneDict` `TypedDict`, but that seems like overkill here.